From 38e454b12c707a0d6870fb5541fad64ece17f78d Mon Sep 17 00:00:00 2001
From: Julian Ospald <hasufell@hasufell.de>
Date: Mon, 17 Nov 2014 22:27:39 +0100
Subject: [PATCH 1/4] Make the "lib" target depend on the "libntru.so" target

Depending on the old "lib" target caused a rebuild of libntru.so on any
such target (e.g. installation), because "lib" is in .PHONY
and not an expected file name.

Fixing this helps with distro policies which require to not do
compilation processes during installation phase.
---
 Makefile.linux | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/Makefile.linux b/Makefile.linux
index b699998..dbc4957 100644
--- a/Makefile.linux
+++ b/Makefile.linux
@@ -22,7 +22,9 @@ MAKEFILENAME=$(lastword $(MAKEFILE_LIST))
 
 all: lib
 
-lib: $(LIB_OBJS_PATHS)
+lib: libntru.so
+
+libntru.so: $(LIB_OBJS_PATHS)
 	$(CC) $(CFLAGS) $(CPPFLAGS) -shared -Wl,-soname,libntru.so -o libntru.so $(LIB_OBJS_PATHS) $(LDFLAGS) $(LIBS)
 
 install: lib
-- 
2.1.3

